packages/website/pages/docs/how-to/retrieve.md

```diff
@@ -2,16 +2,179 @@
-    - Regardless: Store a local copy of your off-chain data
-- If you have your own end-users
-    - Gateways are an easy solution do represent a centralized piece of infrastructure that can provide poor performance under heavy load / worst case can fail
-        - Store a local copy of your off-chain data
-        - If you have your own end users, consider making your own copy of the data available to them
-            - Run your own IPFS node and gateway that stores this data
-            - Store on an HTTP server and keep a database mapping IPFS URIs to these HTTP URIs, so you have both options when serving to users
\ No newline at end of file
+
+import Callout from 'nextra-theme-docs/callout';
+
+In this guide, you'll learn several methods for retrieving NFT data that's been stored with NFT.Storage. 
+
+All data stored using NFT.Storage is made available for retrieval via [IPFS](https://ipfs.io), the InterPlanetary File System. IPFS is a distributed, peer-to-peer network for storing and sharing content-addressed data. 
+
+If you've collected an NFT that uses IPFS, this guide can help you understand how IPFS addresses work and how to view your data using HTTP gateways or native IPFS tools.
+
+You can also find out how to make archival copies of your NFT data using IPFS tools.
+
+<Callout emoji="💡">
+If you're building a platform or marketplace that needs to fetch NFT data on behalf of your end users, please see
+the [architecture considerations guide][concepts-arch-retrieve]
+</Callout>
+
+## Understanding IPFS addresses
+
+IPFS uses a technique called [content addressing][ipfs-docs-concepts-cid] to uniquely identify each piece of data with a short string of characters called a Content ID (CID).
+
+A CID usually looks something like this:
+
+```
+bafybeigvgzoolc3drupxhlevdp2ugqcrbcsqfmcek2zxiw5wctk3xjpjwy
+```
+
+If your NFTs use [IPFS best practices][ipfs-docs-nft-best-practices], the link from the blockchain to your IPFS data will be in the form of a URI that looks like this:
+
+```
+ipfs://bafybeigvgzoolc3drupxhlevdp2ugqcrbcsqfmcek2zxiw5wctk3xjpjwy
+```
+
+This is just a CID with an `ipfs://` prefix added to make it easy to identify as an IPFS CID and not some other kind of random-looking data.
+
+Some IPFS addresses also have a **path** component following the CID, like this:
+
+```
+ipfs://bafybeigvgzoolc3drupxhlevdp2ugqcrbcsqfmcek2zxiw5wctk3xjpjwy/amazing.gif
+```
+
+Here the `/amazing.gif` path points to an image file that's "inside" the IPFS directory with the CID `bafybeigvgzoolc3drupxhlevdp2ugqcrbcsqfmcek2zxiw5wctk3xjpjwy`. 
+
+URLs with the `ipfs://` prefix can be used directly with a browser that supports IPFS like [Brave](https://brave.com). Once you've [configured IPFS support][brave-ipfs], you can just paste an `ipfs://` URL into your address bar and see content in your browser.
```

Comment:
Maybe we should turn those ipfs:// urls into links and instead say that if you are reading this in brave / opera or any other browser that supports ipfs:// URL you can just click those to see the content.